### DISCUSSION TITLE
Link hover effect on in-text links only.

### DISCUSSION THREAD

**Discussion by kulbhushanchand** [mmistakes/minimal-mistakes#2866] [Q&A] [answered]
I'm trying to apply link hover effect as shown in this demo https://codepen.io/pirrera/pen/emKqBp

However, this also effects the links at other places like nav-bar, sidebar etc. What is the best way to apply effects to the links in main body text only?

Thanks

**Comment by iBug** [comment, marked as answer]
Use a better CSS selector like `.page__content a:not(.btn)`. See my code [here][1].

[1]: https://github.com/iBug/iBug-source/blob/7e2db9c3de39aa1e245f8b0334752ee4eb3f8330/_sass/page-content.scss#L17

**Reply by kulbhushanchand** [reply]
That is super helpful. Thanks @iBug 

Here is final code - 
```
.page__content a:not(.btn) {
 
    text-decoration: none;
    border-bottom: 2px solid #ef8b80;
    box-shadow: inset 0 -4px 0 #ef8b80;
    color: inherit;
    transition: background 0.1s cubic-bezier(.33,.66,.66,1);
 
    &:hover {
        background: #ef8b80;
        text-decoration: none;
      }
}
```

**Reply by kulbhushanchand** [reply]
There is little little side-effect with my approach.
The links within the embedded html widgets ([on my website](https://kulbhushanchand.github.io/html-widgets-test/)) also got modified (which is non-desirable), as shown in figure below. 

@iBug Is there any one-size-fit-all solution for all the embedded widgets, or do I have to manually change them for each CSS selector?

![image](https://user-images.githubusercontent.com/17742733/110446511-46e1c780-80e5-11eb-82fd-f9aad366657d.png)

![Untitled](https://user-images.githubusercontent.com/17742733/110447141-eef79080-80e5-11eb-83c8-29985d903338.jpg)

**Update:** I am trying to exclude the `figure` class using `:not(.figure)`, which should exclude all html-widgets from styling, but `plotly` widget seems to be unaffected.

**Reply by iBug** [reply]
Excluding every single widget is surely one way to do this, but as you've probably thought, it's hard to maintain.

An alternative approach is to first determine where your *text* may appear, and then tweak the selector to aim for those text, like this:

```scss
.page__content {
  p, th, td, li {
    > a {
      /* finally here */
    }
  }
}
```

Also, using `a[href]` to select only `<a>` tags that have the `href=` attribute also helps.

**Reply by kulbhushanchand** [reply]
Thanks for the interesting tip.
